allauth/socialaccount/providers/oauth2/views.py

```diff
@@ -128,3 +135,33 @@ def dispatch(self, request):
                 request,
                 self.adapter.provider_id,
                 exception=e)
+
+
+def proxy_login_callback(request, **kwargs):
+    unverified_state = SocialLogin.parse_url_state(request)
+    if 'host' not in unverified_state:
+        raise MissingParameter
+
+    parsed_target = urlparse(unverified_state['host'])
+    target_loc = parsed_target.netloc
+    target_scheme = parsed_target.scheme
+    whitelist_match = False
+    for allowed in app_settings.LOGIN_PROXY_REDIRECT_WHITELIST:
```

Comment:
To shorten this method a bit, might be nice to move this verification into a dedicated method:
```
if not target_in_whitelist(target):
   raise PermissionDenied()
```